src/coreclr/inc/corinfo.h

```diff
@@ -316,6 +316,41 @@ struct SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR
     }
 };
 
+// NOTE:
+// while a struct-arg has no more than two fields and total size is no larger than two-pointer-size.
+// These depends on the platform's ABI rules.
+//
+// `STRUCT_NO_FLOAT_FIELD`=0 means not using the float register(s).
+//
+// the lowest four bits denoting the floating-point info:
+//   bit_0: `1` means there is only one float or double field within the struct.
+//   bit_1: `1` means only the first field is float-point type.
+//   bit_2: `1` means only the second field is float-point type.
+//   bit_3: `1` means the two fields are both float-point type.
+// the bits[5:4] denoting whether the field size is 8-bytes:
+//   bit_4: `1` means the first field's size is 8.
+//   bit_5: `1` means the second field's size is 8.
```

Comment:
> Some editorial suggestions:
> 
> Presumably you could use a 2-bit field with value 1 to mean "there is one floating-point field" and 2 to mean "there are 2 floating-point fields", instead of having both bit 0 and bit 3? Then, value 3 would be invalid, just as you can't set both bit 0 and 3?

yes, can't set both bit 0 and 3.

Reply:
> > Some editorial suggestions:
> > Presumably you could use a 2-bit field with value 1 to mean "there is one floating-point field" and 2 to mean "there are 2 floating-point fields", instead of having both bit 0 and bit 3? Then, value 3 would be invalid, just as you can't set both bit 0 and 3?
> 
> yes, can't set both bit 0 and 3.

Hi, @BruceForstall 
Thanks for your review.

Here using bit 1 and 2 to distinguish what the exact case of the only one float-point field  is needed.
When I save a struct `{float a; int b}` which the float field's offset is 0, while the struct `{int a; float b;}` which the float field's offset is 4.